hs_core/hydroshare/hs_bagit.py

```diff
@@ -222,6 +222,8 @@ def create_bag(resource):
 
     istorage.setAVU(resource.short_id, "resourceType", resource._meta.object_name)
 
+    # delete if there exists any bags for the resource
+    resource.bags.all().delete()
```

Comment:
I only have one minor comment above. Other than that, +1 from me. @pkdash after you double checked and confirmed the build delete() function of the QuerySet object can handle an empty query set, you can merge it.